openml-python-common/src/test/java/com/feedzai/openml/python/ClassificationPythonModelTest.java

```diff
@@ -58,4 +75,35 @@ public final void testSchemaWithoutTargetVariable() {
                 .isInstanceOf(IllegalArgumentException.class);
     }
 
+    /**
+     * Tests that a classifier which classifies an instance with an invalid target value will cause the model to return
+     * NullPointerException.
+     */
+    @Test
+    public final void testInvalidClass() throws URISyntaxException, ExecutionException, InterruptedException {
+        final Path modelPath = Paths.get(getClass().getResource("/dummy_model").toURI());
+        final String id = "classificationModel";
+        final ImmutableList<FieldSchema> fields = ImmutableList.of(FIELD_SCHEMA, CATEGORICAL_FIELD_SCHEMA);
+        final DatasetSchema schema = new DatasetSchema(1, fields);
+        final Random random = new Random();
+
+        this.jepInstance.submitEvaluation(jep -> {
+            // Add the model folder to the python import path
+            jep.eval("import sys");
+            jep.eval(String.format("sys.path.append(\"%s\")", modelPath.toAbsolutePath()));
+
+            // Import the Classifier custom class and store an instance of it in a variable with the name passed in "id"
+            jep.eval("from classifier import Classifier");
+            jep.eval(String.format("%s = Classifier()", id));
+
+            return null;
+        }).get();
```

Comment:
why is this stuff needed?

Reply:
Because I can only trigger the NPE if I have a classifier that returns a class that is outside of the domain of the target field. It may seem a bit overkill, but I think it is the easiest way to had coverage to this class (which was really the achilles heel of the entire project).